server/game/GameActions/AttachAction.js

```diff
@@ -44,7 +44,9 @@ class AttachAction extends CardGameAction {
             { card: this.upgrade, parent: card, context: context },
             (event) => {
                 if (event.card.location === 'play area') {
-                    event.card.parent.removeAttachment(event.card);
+                    if (event.card.parent) {
```

Comment:
I did not understand why this "if" was needed

Reply:
I thought it was needed, but I just took it out and it all works now. I think this was a hold over before I figured out the the "this" problem I was having.